oligo_designer_toolsuite/oligo_specificity_filter/_filter_base.py

```diff
@@ -298,10 +364,130 @@ def _run_filter(
             file_index=file_index,
             region_ids=region_id,
         )
-        table_hits, oligos_with_hits = self._find_hits(
+        table_hits = self._find_hits(
             oligo_database=oligo_database,
             search_results=search_results,
             consider_hits_from_input_region=consider_hits_from_input_region,
         )
+        if self.ai_filter_api is not None:
+            table_hits = self._ai_filter_hits(
+                sequence_type=sequence_type,
+                table_hits=table_hits,
+                file_reference=file_reference,
+                oligo_database=oligo_database,
+                region_id=region_id,
+            )
+
+        return table_hits
+
+    def _ai_filter_hits(
```

Comment:
this whole function is very specific to the binding specificity filter. This should move to an AI filter class.
I.e. it would be good to have a AIfilter base class and then the specific implementations